db.cpp

```diff
@@ -175,22 +175,31 @@ void CAddrDb::GetIPs_(set<CNetAddr>& ips, int max, const bool* nets) {
     } else {
       id = *ourId.begin();
     }
-    if (id >= 0) {
+    if (id >= 0 && idToInfo[id].services & requestedFlags) {
```

Comment:
IMO this is correct. 
` && idToInfo[id].services & requestedFlags` should filter all services that do not have the requestedFlags set (not == but bit AND check). Or am I wrong?